packages/common/container-definitions/src/loader.ts

```diff
@@ -542,6 +542,9 @@ export enum LoaderHeader {
     reconnect = "fluid-reconnect",
     sequenceNumber = "fluid-sequence-number",
 
+    // BREAKING CHANGE -- TODO: figure this out
```

Comment:
Is it a breaking change to add a new enum value? I would expect all usages to be against specific enums but maybe there's something I'm not thinking of.

Reply:
Enums are a pain from a back-compat perspective. Adding a new value creates type incompatibility between the enum type. For example in our type testing code, this blows up if I add an enum value:

```text
Argument of type 'TypeOnly<import("/home/tylerbu/code/release-1/packages/common/container-definitions/src/loader").LoaderHeader>' is not assignable to parameter of type 'TypeOnly<import("/home/tylerbu/code/release-1/node_modules/@fluidframework/container-definitions-previous/dist/loader").LoaderHeader>'.
  Type 'LoaderHeader.cache' is not assignable to type 'TypeOnly<LoaderHeader>'.
```

Reply:
ahh interesting, right, because one definition does not fit into the other. I wonder if we should come up with a general coding recommendation to avoid these

Reply:
Yes, enums are not mapping to native JS. I think we have explored two alternatives: string unions (mostly on string types) and objects pattern. ex: https://github.com/microsoft/FluidFramework/pull/10149/files. I don't think we have concluded on one pattern fits all.